client_internals_it_test.go

```diff
@@ -542,3 +543,176 @@ func ensureClient(config hz.Config) *hz.Client {
 	}
 	panic("the client could not connect to the cluster in 60 seconds.")
 }
+
+type MutualAuthenticationStatus string
+
+const (
+	OPTIONAL MutualAuthenticationStatus = "OPTIONAL"
+	REQUIRED MutualAuthenticationStatus = "REQUIRED"
+)
+
+func TestMutualAuthentication(t *testing.T) {
+	testCases := []struct {
+		name          string
+		muStatus      MutualAuthenticationStatus
+		ca, cert, key string
+		wantErr       bool
+	}{
+		{
+			name:     "testMA_RequiredClientAndServerAuthenticated",
```

Comment:
No need to use the `testMA_` prefix.

Reply:
Could you use the same name as the Java counterpart, like: `whenRequired_andServerNotAuthenticated` and add a comment about where is the test coming from?

Reply:
Added, [c6481ec](https://github.com/hazelcast/hazelcast-go-client/pull/859/commits/c6481ec04328cf044ab3bec8e31bd712bd663bd2).